### PR TITLE
feat: allow filtering by log types

### LINKED ISSUE #1470: Alert Search by Log Type
### Describe the ideal solution

A way to filter alerts by log type

### Describe your use cases

When writing and tuning rules, it's a very common workflow to look at the past alerts for a given log type. This can be accomplished right now as a workaround by running a Data Explorer query.

### References

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/6c4082e9-dc0b-4e26-b60e-cc96e60306ee/commits/latest/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/FDBC5720-EAFE-4F03-92DE-57E59AF01C69?collectionId=674777d0-b802-4e01-9060-36a3a2564744&collectionLayerId=81d2caa5-4aef-4904-b614-7155889e5f39

### PR DESCRIPTION
**Pull request by s0l0ist**
## Background

Closes #1470 

## Changes

- Add backend ddb filter expression
- Add graphql schema field for logTypes
- Update FE to include the logType filter action

## Testing

- mage test:ci
- visually check to see if alerts are properly filtered by log type

Screen Shot 2020-09-24 at 12 23 13 PM

### CODE REVIEW

File: api/lambda/alerts/models/api.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ type ListAlertsInput struct {
 	AlertIDContains *string    `json:"alertIdContains"`
 	EventCountMin   *int       `json:"eventCountMin" validate:"omitempty,min=0"`
 	EventCountMax   *int       `json:"eventCountMax" validate:"omitempty,min=1"`
-
+	LogTypes        []string   `json:"logTypes" validate:"omitempty"`
```
Comment (austinbyers) [inline api/lambda/alerts/models/api.go:94]: `validate:"omitempty"` is equivalent to having nothing at all. Maybe you want

```go
validate:"omitempty,dive,required" // ensures no empty strings in the slice
```

**Review by austinbyers — APPROVED** [review]
Nice!

File: api/graphql/schema.graphql [hunk 2]
```diff
@@ -104,6 +104,7 @@ input ListAlertsInput {
   status: [AlertStatusesEnum]
   eventCountMin: Int
   eventCountMax: Int
+  logTypes: [String]
```
Comment (3nvi) [inline api/graphql/schema.graphql:107]:
```suggestion
  logTypes: [String!]
```

`logTypes` can be omitted, but if it's not, then it should be containing strings

File: api/lambda/alerts/models/api.go (hunk 1 above)
Comment (3nvi) [reply api/lambda/alerts/models/api.go:94]: agree. Same thing with the schema above 😄

File: web/src/pages/ListAlerts/ListAlertsActions/ListAlertsActions.tsx [hunk 3]
```diff
@@ -53,6 +54,15 @@ export const filters = {
       placeholder: 'Choose a status...',
     },
   },
+  logTypes: {
```
Comment (3nvi) [inline web/src/pages/ListAlerts/ListAlertsActions/ListAlertsActions.tsx:57]: @vorillaz do we plan to change this soon?

**Review by 3nvi — APPROVED** [review]

**Comment by jacknagz** [comment]
This is going to be so helpful!

**Pull request merged**